_studio/shared/umc/codec/av1_dec/src/umc_av1_va_packer_vaapi.cpp

```diff
@@ -0,0 +1,391 @@
+
+            fg.grain_seed = (uint16_t)info.film_grain_params.grain_seed;
+            fg.num_y_points = (uint8_t)info.film_grain_params.num_y_points;
+
+            for (uint8_t i = 0; i < MAX_POINTS_IN_SCALING_FUNCTION_LUMA; i++)
+            {
+                fg.point_y_value[i] = (uint8_t)info.film_grain_params.point_y_value[i];
+                fg.point_y_scaling[i] = (uint8_t)info.film_grain_params.point_y_scaling[i];
+            }
+
+            fg.num_cb_points = (uint8_t)info.film_grain_params.num_cb_points;
+            fg.num_cr_points = (uint8_t)info.film_grain_params.num_cr_points;
+
+            for (uint8_t i = 0; i < MAX_POINTS_IN_SCALING_FUNCTION_CHROMA; i++)
+            {
+                fg.point_cb_value[i] = (uint8_t)info.film_grain_params.point_cb_value[i];
+                fg.point_cb_scaling[i] = (uint8_t)info.film_grain_params.point_cb_scaling[i];
+                fg.point_cr_value[i] = (uint8_t)info.film_grain_params.point_cr_value[i];
+                fg.point_cr_scaling[i] = (uint8_t)info.film_grain_params.point_cr_scaling[i];
+            }
+
+            for (uint8_t i = 0; i < MAX_AUTOREG_COEFFS_LUMA; i++)
+                fg.ar_coeffs_y[i] = (int8_t)info.film_grain_params.ar_coeffs_y[i];
+
+            for (uint8_t i = 0; i < MAX_AUTOREG_COEFFS_CHROMA; i++)
+            {
+                fg.ar_coeffs_cb[i] = (int8_t)info.film_grain_params.ar_coeffs_cb[i];
+                fg.ar_coeffs_cr[i] = (int8_t)info.film_grain_params.ar_coeffs_cr[i];
+            }
+
+            fg.cb_mult = (uint8_t)info.film_grain_params.cb_mult;
+            fg.cb_luma_mult = (uint8_t)info.film_grain_params.cb_luma_mult;
+            fg.cb_offset = (uint8_t)info.film_grain_params.cb_offset;
+            fg.cr_mult = (uint8_t)info.film_grain_params.cr_mult;
+            fg.cr_luma_mult = (uint8_t)info.film_grain_params.cr_luma_mult;
+            fg.cr_offset = (uint8_t)info.film_grain_params.cr_offset;
+        }
+
+        // fill tile params
+        picParam.tile_cols = (uint8_t)info.tile_info.TileCols;
+        picParam.tile_rows = (uint8_t)info.tile_info.TileRows;
+
+        for (uint32_t i = 0; i < picParam.tile_cols; i++)
+        {
+            picParam.width_in_sbs_minus_1[i] =
+                (uint16_t)(info.tile_info.SbColStarts[i + 1] - info.tile_info.SbColStarts[i] - 1);
+        }
+
+        for (int i = 0; i < picParam.tile_rows; i++)
+        {
+            picParam.height_in_sbs_minus_1[i] =
+                (uint16_t)(info.tile_info.SbRowStarts[i + 1] - info.tile_info.SbRowStarts[i] - 1);
+        }
+
+        picParam.context_update_tile_id = (uint16_t)info.tile_info.context_update_tile_id;
+        // TODO: [Rev0.85] Add correct setting for large scale tile mode
+        picParam.tile_count_minus_1 = 0;
+    }
+
+    void PackerVA::PackTileControlParams(VASliceParameterBufferAV1& tileControlParam, TileLocation const& loc)
+    {
+#if VA_CHECK_VERSION(1,7,0)
```

Comment:
Please, remove this check: va_av1_dec.h header first appeared in VAAPI 1.7 and this if is pointless. 
* https://github.com/intel/libva/blob/v2.6-branch/va/va_dec_av1.h
* https://github.com/intel/libva/blob/v2.7-branch/va/va_dec_av1.h

Reply:
sure